spec/controllers/equipment_models_controller_spec.rb

```diff
@@ -1,421 +1,323 @@
           expect(assigns(:equipment_model)).to eq(model)
         end
-        it 'updates attributes' do
-          model.reload
-          expect(model.name).to eq('Mod')
+        it 'sets @associated_equipment_models' do
+          mod1 = FactoryGirl.create(:equipment_model)
+          model.associated_equipment_models = [mod1]
+          get :show, id: model
+          expect(assigns(:associated_equipment_models)).to eq([mod1])
+        end
+
+        it 'limits @associated_equipment_models to maximum 6' do
+          model.associated_equipment_models =
+            FactoryGirl.create_list(:equipment_model, 7)
+          get :show, id: model
+          expect(assigns(:associated_equipment_models).size).to eq(6)
         end
-        it { is_expected.to redirect_to(model) }
       end
+      USER_ROLES.each { |type| it_behaves_like 'GET show success', type }
 
-      context 'without valid attributes' do
+      context 'with admin user' do
         before do
-          put :update, id: model, equipment_model:
-          FactoryGirl.attributes_for(:equipment_model, name: nil)
+          FactoryGirl.create_pair(:equipment_item, equipment_model: model)
+          mock_user_sign_in(mock_user(:admin))
         end
-        it { is_expected.not_to set_flash }
-        it 'should not update attributes' do
-          model.reload
-          expect(model.name).not_to be_nil
+        let!(:missed) do
+          FactoryGirl.create(:missed_reservation, equipment_model: model)
+        end
+        let!(:starts_today) do
+          FactoryGirl.create(:reservation, equipment_model: model,
+                                           start_date: Time.zone.today,
+                                           due_date: Time.zone.today + 2.days)
+        end
+        let!(:starts_this_week) do
+          FactoryGirl.create(:reservation, equipment_model: model,
+                                           start_date: Time.zone.today + 2.days,
+                                           due_date: Time.zone.today + 4.days)
+        end
+        let!(:starts_next_week) do
+          FactoryGirl.create(:reservation, equipment_model: model,
+                                           start_date: Time.zone.today + 9.days,
+                                           due_date: Time.zone.today + 11.days)
+        end
+        it 'includes @pending reservations' do
+          get :show, id: model
+          expect(assigns(:pending)).to \
+            match_array([starts_today, starts_this_week])
         end
-        it { is_expected.to render_template(:edit) }
       end
-      it 'calls delete_files'
-    end
-    context 'with non-admin user' do
-      before { sign_in FactoryGirl.create(:user) }
-      it 'should redirect to root' do
-        put :update,
-            id: model,
-            equipment_model: FactoryGirl.attributes_for(:equipment_model)
-        expect(response).to redirect_to(root_url)
+
+      context 'availability calendar' do
```

Comment:
while testing availability calculations are important, this kind of tests really belong in the model spec. We should make sure that the model spec adequately tests the availability methods and remove these.